airbyte-integrations/connectors/destination-snowflake/src/test-integration/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGeneratorIntegrationTest.java

```diff
@@ -369,4 +372,24 @@ record -> record.get(JavaBaseConstants.COLUMN_NAME_AB_RAW_ID).asText(),
     });
   }
 
+  /**
+   * We test this for Snowflake because we made a special exception class, _ab_missing_primary_key
+   */
+  @Test
+  protected void customExceptionWhenMissingPK() throws Exception {
+    createRawTable(streamId);
+    createFinalTable(incrementalDedupStream, "");
+    insertRawTableRecords(
+        streamId,
+        BaseTypingDedupingTest.readRecords("sqlgenerator/incrementaldedup_inputrecords_missing_pk.jsonl"));
+
+    final String sql = generator.updateTable(incrementalDedupStream, "");
+    Exception exception = assertThrows(Exception.class, () -> { // TODO (evan): I don't really know what class of error the Snowflake driver throws
```

Comment:
run it in a debugger and see? (I can do this if your intellij is sad, it's probably a jdbc exception)